From 163ef20905bbb30172aee99e9b32211cc5282652 Mon Sep 17 00:00:00 2001
From: Rajeev Surampally <rsurampally@cavium.com>
Date: Wed, 12 Dec 2012 19:26:55 -0800
Subject: [PATCH 273/337] netdev: octeon-ethernet: Export port and pool config
 to named block.

Based On SDK 3.0.0-482

This named block will be used by cooperating apps to import config
and use them.

Signed-off-by: Rajeev Surampally <rsurampally@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/cavium-octeon/executive/Makefile |    2 +-
 drivers/net/ethernet/octeon/ethernet.c     |    5 +++++
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/arch/mips/cavium-octeon/executive/Makefile b/arch/mips/cavium-octeon/executive/Makefile
index 56ffe16..9c19fc6 100644
--- a/arch/mips/cavium-octeon/executive/Makefile
+++ b/arch/mips/cavium-octeon/executive/Makefile
@@ -25,7 +25,7 @@ obj-y += cvmx-interrupt-decodes.o cvmx-interrupt-rsl.o
 obj-y += cvmx-helper-errata.o cvmx-helper-jtag.o
 
 obj-y += cvmx-global-resources.o cvmx-range.o \
-	cvmx-pko-internal-ports-range.o
+	cvmx-pko-internal-ports-range.o cvmx-appcfg-transport.o
 
 obj-y += cvmx-clock.o octeon-feature.o
 
diff --git a/drivers/net/ethernet/octeon/ethernet.c b/drivers/net/ethernet/octeon/ethernet.c
index 5c15eb2..846d71e 100644
--- a/drivers/net/ethernet/octeon/ethernet.c
+++ b/drivers/net/ethernet/octeon/ethernet.c
@@ -48,6 +48,7 @@
 #include <asm/octeon/cvmx-ipd.h>
 #include <asm/octeon/cvmx-srio.h>
 #include <asm/octeon/cvmx-helper.h>
+#include <asm/octeon/cvmx-app-config.h>
 
 #include <asm/octeon/cvmx-gmxx-defs.h>
 #include <asm/octeon/cvmx-smix-defs.h>
@@ -311,6 +312,8 @@ static __devinit int cvm_oct_configure_common_hw(void)
 	/* more configuration needs to be done, so enable ipd seperately */
 	cvmx_ipd_cfg.ipd_enable = 0;
 
+	__cvmx_export_app_config_to_named_block(CVMX_APP_CONFIG);
+
 	cvmx_helper_initialize_packet_io_global();
 
 #ifdef __LITTLE_ENDIAN
@@ -1136,6 +1139,8 @@ static int __devexit cvm_oct_remove(struct platform_device *pdev)
 
 	cvm_oct_mem_cleanup();
 
+	__cvmx_export_app_config_cleanup();
+
 	return 0;
 }
 
-- 
1.7.5.4

